js/HintView.js

```diff
@@ -0,0 +1,57 @@
+import Adapt from 'core/js/adapt';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { templates } from 'core/js/reactHelpers';
+import notify from 'core/js/notify';
+
+export default class HintView extends Backbone.View {
+
+    className() {
+        return 'hint';
+    }
+
+    initialize() {
+        this.onHintPopupClicked = this.onHintPopupClicked.bind(this);
+
+        this.listenTo(Adapt, 'remove', this.remove);
+
+        this.render();
+    }
+
+    render() {
+        const data = this.model.toJSON();
+        
+        //Check if the component has a header to determine where the hint element will be created
+        const displayTitle = this.model.get('displayTitle');
+        const body = this.model.get('body');
+        const instruction = this.model.get('instruction');
+
+        const containsHeader = displayTitle || body || instruction;
+        const componentHeaderInner = $('.' + this.model.get('_id')).find('.component__header-inner');
+        const componentWidget = $('.' + this.model.get('_id')).find('.component__widget');
+        const targetSelector = containsHeader ? componentHeaderInner  : componentWidget;        
```

Comment:
The is not a selector, it's a jQuery object with the found elements in it. 
```suggestion
        const $target = containsHeader ? componentHeaderInner  : componentWidget;        
```